File: youlexuan-goods-interface/src/main/java/com/offcn/goods/service/TypeService.java
package com.offcn.goods.service;

import com.offcn.pojo.TbTypeTemplate;
import com.offcn.vo.resp.PageResult;

import java.util.List;
import java.util.Map;

public interface TypeService {

    /**
     * 获取所有的模板列表
     * @return
     */
    public List<TbTypeTemplate> typeList();

    /**
     * 分页获取所有模板
     * @param pageNum
     * @param pageSize
     * @param tbTypeTemplate
     * @return
     */
    public PageResult search(int pageNum,int pageSize,TbTypeTemplate tbTypeTemplate);

    /**
     * 通过id找到模块对象
     * @param id
     * @return
     */
    public TbTypeTemplate findOne(Long id);

    /**
     * 修改
     * @param tbTypeTemplate
     */
    public void update(TbTypeTemplate tbTypeTemplate);

    /**
     * 添加
     * @param tbTypeTemplate
     */
    public void add(TbTypeTemplate tbTypeTemplate);

    /**
     * 通过id删除
     * @param id
     */
    public void delete(Long id);

    /**
     * 批量删除
     * @param ids
     */
    public void deleteByIds(Long[] ids);

    /**
     * 获取所有的brand和spec list集合的map集合
     * @return
     */
    public Map getBrandAndSpecList();

    /**
     * 根据模板id获取规格和规格选项
     * @param templateId
     * @return
     */
    public List<Map> getSpecAndOptionList(Long templateId);
}

File: youlexuan-manage-web/src/main/webapp/js/service/typeService.js
app.service('typeService',function ($http) {

    //分页查询
    this.search=function (url,entity) {

        return $http.post(url,entity);
    }

    //编辑回显一个对象
    this.findOne=function (id) {

        return $http.post("/type/findOne.do?id="+id);
    }

    //获取所有品牌和规格
    this.getBrandAndSpecList=function () {

        return $http.get("/type/getBrandAndSpec.do");
    }

    //修改
    this.update=function (entity) {

        return $http.post("/type/update.do",entity);
    }

    //添加
    this.add=function (entity) {

        return $http.post("/type/add.do",entity);
    }

    //删除
    this.deleteByIds=function (ids) {

        return $http.post("/type/deleteByIds.do?ids="+ids);
    }
});